addons/l10n_sa_edi/models/account_edi_format.py

```diff
@@ -436,11 +436,13 @@ def _set_missing_partner_fields(missing_fields, name):
             errors.append(_set_missing_partner_fields(supplier_missing_info, _("Supplier")))
         if customer_missing_info:
             errors.append(_set_missing_partner_fields(customer_missing_info, _("Customer")))
+
         user_timezone = pytz.timezone(self.env.user.tz or 'utc')
         invoice_datetime = datetime.combine(invoice.invoice_date, datetime.min.time())
         user_invoice_datetime = user_timezone.localize(invoice_datetime)
-        utcnow = pytz.utc.localize(datetime.now())
-        if user_invoice_datetime > utcnow:
+        invoice_date_utc = user_invoice_datetime.astimezone(pytz.timezone('utc'))
+        # We should compare dates instead of datetime as it will result in inequalities.
+        if invoice_date_utc.date() > date.today():
```

Comment:
```suggestion
        if invoice.invoice_date > fields.Date.context_today(invoice):
```
This looks like an overly complicated logic to compare 2 accounting dates :thinking: 
I think you could simply do this instead (and remove several unnecessary variables), no?
Or maybe I'm missing something :sweat_smile: